packages/binding-http/src/codecs/tuya-codec.ts

```diff
@@ -17,17 +17,28 @@ import { ContentCodec } from "@node-wot/core";
 import * as TD from "@node-wot/td-tools";
 import { DataSchemaValue } from "wot-typescript-definitions";
 
+interface TuyaOutput {
+    success?: boolean;
+    msg?: string;
+    result?: {
+        code?: string;
+    }[];
+}
+
 export default class HttpTuyaCodec implements ContentCodec {
     getMediaType(): string {
         return "application/json+tuya";
     }
 
     bytesToValue(bytes: Buffer, schema: TD.DataSchema, parameters: { [key: string]: string }): DataSchemaValue {
-        const parsedBody = JSON.parse(bytes.toString());
-        if (!parsedBody.success) throw new Error(parsedBody.msg ? parsedBody.msg : JSON.stringify(parsedBody));
-        for (const key in parsedBody.result) {
-            if (parsedBody.result[key].code === schema["tuya:PropertyName"]) {
-                return parsedBody.result[key].value;
+        const parsedBody: TuyaOutput = JSON.parse(bytes.toString());
+        if (parsedBody.success !== true) {
```

Comment:
is this allowed?
```suggestion
        if (!(parsedBody.success ?? false)) {
```

[source](https://typescript-eslint.io/rules/strict-boolean-expressions/): See last correct example

Reply:
I think both versions are fine here – in the end, it is probably a question of readability 🤔 Due to otherwise doubled negation, I actually tend to the `!== true` version in this instance but I would also be fine with nullish coalescing :)

Reply:
In order to maximize readability, I now introduced an `success` auxiliary variable – so I think, this should now be resolved :)